src/qbe_ir_generator.cpp

```diff
@@ -151,11 +151,32 @@ void QbeIrGenerator::Visit(const DeclVarNode& decl) {
   id_to_num[decl.id] = id_num;
 }
 
-void QbeIrGenerator::Visit(const DeclArrNode& array_decl) {
-  int id_num = NextLocalNum();
-  WriteInstr_("{} =l alloc{} {}", FuncScopeTemp{id_num},
-              array_decl.type->element_type().size(), array_decl.type->size());
-  id_to_num[array_decl.id] = id_num;
+void QbeIrGenerator::Visit(const DeclArrNode& arr_decl) {
+  int base_addr_num = NextLocalNum();
+  std::size_t element_size = arr_decl.type->element_type().size();
+  WriteInstr_("{} =l alloc{} {}", FuncScopeTemp{base_addr_num}, element_size,
+              arr_decl.type->size());
+  id_to_num[arr_decl.id] = base_addr_num;
+
+  std::size_t index = 0;
+  for (auto& arr_init : arr_decl.init_list) {
+    arr_init->Accept(*this);
```

Comment:
Since we're iterating over `index` and `arr_decl.init_list` concurrently, it would be better to specify the index in the loop header and access `arr_decl.init_list` inside the loop. This approach ensures that the lifetime of `index` remains within the loop and eliminates the need for manual increments.
```suggestion
  for (auto i = std::size_t{0}, e = arr_decl.init_list->size(); i < e; ++i) {
    auto& arr_init = arr_decl.init_list.at(i);
    arr_init->Accept(*this);
```